src/java.desktop/share/classes/java/awt/Toolkit.java

```diff
@@ -1742,10 +1742,9 @@ public void removeAWTEventListener(AWTEventListener listener) {
 
         synchronized (this) {
             SelectiveAWTEventListener selectiveListener =
-                listener2SelectiveListener.get(localL);
+                listener2SelectiveListener.remove(localL);
```

Comment:
While analyzing the implementation of this method, I noticed a potential bug at the beginning. Should it instead be:
```
if (localL == null) {
    return;
}
```
?

Reply:
Looks like yes. I think it's worth to create a separate issue to fix it.

Reply:
It doesn't matter, it looks… `localL` is `null` if and only if listener is `null`.

However, I'd put the null-check above the call to `deProxyAWTEventListener` to avoid any confusion.

Reply:
listener and proxy are public classes and there's no assertion that EventListenerProxy.getListener() will always return a non-null value. So this method should fetch the listener from the proxy and check it for null similar to how it is implemented in addAWTEventListener. Currently, it works fine because we never insert null values into the map. so it is just a code clarification.

Reply:
You're right!